File: app/src/main/java/com/aandssoftware/aandsinventory/models/CarouselMenuModel.java

package com.aandssoftware.aandsinventory.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class CarouselMenuModel extends RealmObject {
  
  
  @SerializedName("id")
  @Expose
  @PrimaryKey
  private int id;
  
  @SerializedName("aliceName")
  @Expose
  private String aliceName;
  
  @SerializedName("imageId")
  @Expose
  private String imageId;
  
  @SerializedName("expression")
  @Expose
  private String expression;
  
  @SerializedName("tag")
  @Expose
  private String tag;
  
  @SerializedName("description")
  @Expose
  private String description;
  
  @SerializedName("dateCreated")
  @Expose
  private long dateCreated;
  
  @SerializedName("CarouselId")
  @Expose
  private String CarouselId;
  
  public CarouselMenuModel() {
  
  }
  
  public CarouselMenuModel(CarouselMenuModel item) {
    this.id = item.getId();
    this.aliceName = item.getAliceName();
    this.imageId = item.getImageId();
    this.aliceName = item.getAliceName();
    this.expression = item.getExpression();
    this.tag = item.getTag();
    this.description = item.getDescription();
    this.dateCreated = item.getDateCreated();
    this.CarouselId = item.getCarouselId();
  }
  
  public int getId() {
    return id;
  }
  
  public CarouselMenuModel setId(int id) {
    this.id = id;
    return this;
  }
  
  public String getAliceName() {
    return aliceName;
  }
  
  public CarouselMenuModel setAliceName(String aliceName) {
    this.aliceName = aliceName;
    return this;
  }
  
  public String getImageId() {
    return imageId;
  }
  
  public CarouselMenuModel setImageId(String imageId) {
    this.imageId = imageId;
    return this;
  }
  
  public String getExpression() {
    return expression;
  }
  
  public CarouselMenuModel setExpression(String expression) {
    this.expression = expression;
    return this;
  }
  
  public String getTag() {
    return tag;
  }
  
  public CarouselMenuModel setTag(String tag) {
    this.tag = tag;
    return this;
  }
  
  public String getDescription() {
    return description;
  }
  
  public CarouselMenuModel setDescription(String description) {
    this.description = description;
    return this;
  }
  
  public long getDateCreated() {
    return dateCreated;
  }
  
  public CarouselMenuModel setDateCreated(long dateCreated) {
    this.dateCreated = dateCreated;
    return this;
  }
  
  public String getCarouselId() {
    return CarouselId;
  }
  
  public CarouselMenuModel setCarouselId(String carouselId) {
    CarouselId = carouselId;
    return this;
  }
}

File: app/src/main/java/com/aandssoftware/aandsinventory/models/CarouselMenuModel.kt
package com.aandssoftware.aandsinventory.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class CarouselMenuModel : Serializable {

    @SerializedName("id")
    @Expose
    var id: Int = 0

    @SerializedName("aliceName")
    @Expose
    var aliceName: String? = null

    @SerializedName("imageId")
    @Expose
    var imageId: String? = null

    @SerializedName("expression")
    @Expose
    var expression: String? = null

    @SerializedName("tag")
    @Expose
    var tag: String? = null

    @SerializedName("description")
    @Expose
    var description: String? = null

    @SerializedName("dateCreated")
    @Expose
    var dateCreated: Long = 0

    @SerializedName("carouselId")
    @Expose
    var carouselId: String? = null

    @SerializedName("permissions")
    @Expose
    var permissions: String? = null

    @SerializedName("defaultImageId")
    @Expose
    var defaultImageId: Int = 0

    companion object {
        const val TABLE_CAROUSEL = "carousel"
    }
}
